eval/builtin_fn_styled.go

```diff
@@ -0,0 +1,155 @@
+package eval
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elves/elvish/eval/vals"
+	"github.com/elves/elvish/eval/vars"
+	"github.com/elves/elvish/styled"
+	"github.com/xiaq/persistent/hashmap"
+)
+
+var styledTransformers hashmap.Map
+
+func init() {
+	addBuiltinFns(map[string]interface{}{
+		"styled-segment": styledSegment,
+		"styled":         StyledBuiltin,
+	})
+
+	transformers := make(map[interface{}]interface{})
+	for k, v := range styled.SegmentTransformers {
+		transformers[k] = NewBuiltinFn("style-transform "+k, v)
```

Comment:
These builtins are not actually accessible from the `builtin:` namespace. Give them more funny names, like `<style transformer xxx>`.

Speaking of which, I thought a bit more about the API, and I think that exposing transformers as a writable map is likely a bad idea: this is essentially a global variable, and it is hard to coordinate access from different modules written by different people. For instance, one module may install a transformer named `foo` and use it later, while another may install a transformer *with the same name* `foo`. One of the two modules will use a transformer that it does not intend to use.

My amended proposal is to not expose this transformer map at all and keep it strictly internal. This part of the code can be removed.

In fact, after removing the transformer map, the builtin transformers do not even need to live in a map; the `styled` command can also parse the transformer programmatically.